controls/V-259740.rb

```diff
@@ -0,0 +1,81 @@
+  desc 'check', 'If new packages are available for PostgreSQL, they can be reviewed in the package manager appropriate for the server operating system:
+
+To list the version of installed PostgreSQL using psql:
+
+$ sudo su - postgres
+$ psql --version
+
+To list the current version of software for RPM:
+
+$ rpm -qa | grep postgres
+
+To list the current version of software for APT:
+
+$ apt-cache policy postgres
+
+All versions of PostgreSQL will be listed here:
+http://www.postgresql.org/support/versioning/
+
+All security-relevant software updates for PostgreSQL will be listed here:
+http://www.postgresql.org/support/security/
+
+If PostgreSQL is not at the latest version, this is a finding.'
+  desc 'fix', 'Remove or decommission all unsupported software products.
+
+Upgrade unsupported DBMS or unsupported components to a supported version of the product.'
+  impact 0.7
+  tag check_id: 'C-54617r944424_chk'
+  tag severity: 'high'
+  tag gid: 'V-259740'
+  tag rid: 'SV-259740r944425_rule'
+  tag stig_id: 'CD12-00-012900'
+  tag gtitle: 'SRG-APP-000456-DB-000400'
+  tag fix_id: 'F-54571r944425_fix'
+  tag 'documentable'
+  tag cci: ['CCI-003376']
+  tag nist: ['SA-22 a']
+
+  postgres_changelog = inspec.http('https://www.postgresql.org/support/versioning/').body.lines.map(&:chomp)
+  postgres_changelog_supported = postgres_changelog.select{ |line| line.include? 'Yes' }
+
+  supported_lines = postgres_changelog.each_index.select { |i| postgres_changelog[i].include? 'Yes' }
+  supported_indices = supported_lines.map{ |i| i - 1 }
+
+  supported_versions = supported_indices.map { |i| postgres_changelog[i] }
+  supported_versions = supported_versions.map { |string| string.scan(/\d+\.\d+/).first }
+
+  min_org_allowed_postgres_version = input('min_org_allowed_postgres_version')
+
+  installed_postgres_version = command('psql --version').stdout.split[2]
+
+  # Extract major release version numbers for first test
+  supported_versions_major = supported_versions.map { |i| i.split('.').first }
+  installed_postgres_version_major = installed_postgres_version.split('.').first
+
+  # If no organization specified postgres version was given, check the internet for major and minor release versions
+  if (min_org_allowed_postgres_version.nil? || min_org_allowed_postgres_version.empty?)
+    describe "PostgreSQL installed major release" do
+      subject { installed_postgres_version_major }
+      it { should be_in supported_versions_major }
+    end
+    if supported_versions_major.include?(installed_postgres_version_major)
+      allowed_version_full = supported_versions.find{ |i| i.split('.').first == installed_postgres_version_major }
+      describe "PostgreSQL installed full version" do
+        subject { installed_postgres_version }
+        it { should eq allowed_version_full }
+      end
+    end
+  else
```

Comment:
Wonder if we want this as the default case because there are probably instances where the organization perhaps wants to make sure the version is at least 16.2, for example, but maybe they are on a lower, but supported vendor version. Then, they would pass even though they don't have 16.2. So maybe flip the logic? Or maybe there is a better overall strategy to think this through? @aaronlippold thoughts?